processing/src/main/java/io/druid/query/Query.java

```diff
@@ -65,6 +66,8 @@
 
   boolean hasFilters();
 
+  DimFilter getFilter();
```

Comment:
IMO we have a long and proud history of `null` return for various Query things, and this is not the right PR to change those to Optional.

Also I'm not totally sure Guava Optional actually makes the code significantly better, it's missing stuff like `foreach` / `ifPresent`, `map`, and `flatMap` that make Scala Options and Java 8 Optionals really usable.